c_models/makefiles/extra.mk

```diff
@@ -0,0 +1,148 @@
+
+# ----------------------------------------------------------------------
+# For multiple source directories, see note below.
+# Make sure EXTRA_SRC_DIR points to the source directory where your unmodified
+# files are found.
+#
+EXTRA_SRC_DIR := $(abspath $(dir $(EXTRA_MAKEFILE_PATH))/../src)
+
+# ----------------------------------------------------------------------
+# We need another directory where the converted C code will go.
+#
+EXTRA_MODIFIED_DIR := $(abspath $(dir $(EXTRA_MAKEFILE_PATH))/../modified_src)
+
+# ----------------------------------------------------------------------
+# We need to point where the logging dictionary mappings will be stored.
+#
+EXTRA_LOG_DICT_FILE := $(EXTRA_MODIFIED_DIR)/log_dict.dict
+
+# ----------------------------------------------------------------------
+# Add the log_file to the list to ensure that they are brought up to date at
+# the right stage of any build. It also ensures that they are added to the
+# final app dictionary file.
+#
+LOG_DICT_FILES += $(LOG_DICT_FILE)
+
+# ----------------------------------------------------------------------
+# We need to tell the compliler to use the modified code.
+#
+CFLAGS += -I$(EXTRA_MODIFIED_DIR)
+
+# ----------------------------------------------------------------------
+# Make sure each neuron model has a unique build directory.
+#
+BUILD_DIR := $(dir $(EXTRA_MAKEFILE_PATH)))/../build/$(APP)
+
+# ----------------------------------------------------------------------
+# Support function for converting original source paths to modified paths.
+#
+# $(NEURAL_MODELLING_DIRS)/src is done by neural_build.mk so not needed here.
+# This function, if defined, is called from within convert_to_modified.
+#
+define extra_convert_to_modified
+$(patsubst $(EXTRA_SRC_DIR)%,$(EXTRA_MODIFIED_DIR)%,$(1))
+endef
+
+# ----------------------------------------------------------------------
+# Support function for converting modified C paths to object paths.
+#
+# $(NEURAL_MODELLING_DIRS)..... is done by neural_build.mk so not needed here.
+# This function, if defined, is called from within convert_to_object.
+#
+define extra_convert_to_object
+$(patsubst $(EXTRA_MODIFIED_DIR)%.c, $(BUILD_DIR)%.o, $(1))
+endef
+
+# ----------------------------------------------------------------------
+
+include $(NEURAL_MODELLING_DIRS)/makefiles/neuron/neural_build.mk
+
+# ----------------------------------------------------------------------
+
+# EXTRA_SYNAPSE_TYPE_OBJECTS is deprecated!
+# Instead add a new rule
+#$(BUILD_DIR).....o: $(EXTRA_MODIFIED_DIR)...c $(C_FILES_MODIFIED)
+#	-mkdir -p $(dir $@)
+#	$(SYNAPSE_TYPE_COMPILE) -o $@ make$<
+
+                       
+# EXTRA_STDP is deprecated
+# Instead add a new rule
+#$(BUILD_DIR).....o: $(EXTRA_MODIFIED_DIR)...c $(C_FILES_MODIFIED)
+#	-mkdir -p $(dir $@)
+#	$(STDP_COMPILE) -o $@ $<
+
+# ----------------------------------------------------------------------
+# Make rules to copy and transform the extra code (so that logging messages are
```

Comment:
i wouldnt be touching these./ why not hidden?

Reply:
Support for people who can read make files

Reply:
ermmm so the 1% of the end users we have. why not make it usable for the 99% of the end users we have, who dont care?

Reply:
This is obsoleted by my changes anyway - you now no longer specify these things, and just add your directory to SOURCE_DIRS, and the system handles the rest.

Reply:
should this pr be closed then? if yours is building upon it?

Reply:
My changes are already in this PR - you may need to refresh to see them